include/tbl/cli_cmds.h

```diff
@@ -123,6 +127,20 @@ CLI_CMD(VCL_LIST,
 	0, 0
 )
 
+CLI_CMD(VCL_DEPS,
```

Comment:
I am not sure how useful this is. For the tabular format, doesn't `vcl.list` already give the relevant information?
For the JSON format, should we actually have a new command or rather extend `vcl.list -j` in a backwards compat manner?

Reply:
> I am not sure how useful this is.

I know, I initially wrote `vcl.deps` first until it lead me to a dead end for my use case. I kept it in hindsight to make it convenient to visualize dependencies. I didn't want to generate a dot graph directly so I also kept the standard `tsort(1)` format.

> For the tabular format, doesn't `vcl.list` already give the relevant information?

No, it tells you that your VCL is labelled, and that the label is used in a `return(vcl)` transition.

> For the JSON format, should we actually have a new command or rather extend `vcl.list -j` in a backwards compat manner?

I find it harder to leverage a JSON output in this case rather than the plain text counterpart. But I have a strong shell bias.

Reply:
> No, it tells you that your VCL is labelled, and that the label is used in a return(vcl) transition.

There must be something obvious which I am missing. Do you have a side-by-side example with `vcl.list` and `vcl.deps` illustrating your case?

Reply:
I made this change:

```diff
--- bin/varnishtest/tests/c00077.vtc
+++ bin/varnishtest/tests/c00077.vtc
@@ -64,4 +64,6 @@ varnish v1 -clierr 106 "vcl.label vclA vcl3"
 
 varnish v1 -cliok "vcl.symtab"
 
+varnish v1 -cliok "vcl.list"
+varnish v1 -cliok "vcl.deps"
 varnish v1 -cliok "vcl.discard vcl1 vcl2 vcl3 vcl4 vcl5 vcl6 vcl7 vclA vclB"
```

This is the output:

```
**   top  === varnish v1 -cliok "vcl.list"
**** v1   CLI TX|vcl.list
***  v1   CLI RX  200
**** v1   CLI RX|available   auto     warm         0    vcl1    <-    (1 label)
**** v1   CLI RX|available   label    warm         0    vclA    ->    vcl1 (5 return(vcl)'s)
**** v1   CLI RX|available   auto     warm         0    vcl2    <-    (1 label)
**** v1   CLI RX|available   label    warm         0    vclB    ->    vcl2 (1 return(vcl))
**** v1   CLI RX|available   auto     warm         0    vcl3
**** v1   CLI RX|available   auto     warm         0    vcl4
**** v1   CLI RX|available   auto     warm         0    vcl5
**** v1   CLI RX|available   auto     warm         0    vcl6
**** v1   CLI RX|available   auto     warm         0    vcl7
**** v1   CLI RX|active      auto     warm         0    vcl8
**   v1   CLI 200 <vcl.list>
**   top  === varnish v1 -cliok "vcl.deps"
**** v1   CLI TX|vcl.deps
***  v1   CLI RX  200
**** v1   CLI RX|vcl1
**** v1   CLI RX|vclA   vcl1
**** v1   CLI RX|vcl2   vclA
**** v1   CLI RX|vclB   vcl2
**** v1   CLI RX|vcl3   vclB
**** v1   CLI RX|vcl4   vclA
**** v1   CLI RX|vcl5   vclA
**** v1   CLI RX|vcl6   vclA
**** v1   CLI RX|vcl7   vclA
**** v1   CLI RX|vcl8
**   v1   CLI 200 <vcl.deps>
**   top  === varnish v1 -cliok "vcl.discard vcl1 vcl2 vcl3 vcl4 vcl5 vcl6...
**** v1   CLI TX|vcl.discard vcl1 vcl2 vcl3 vcl4 vcl5 vcl6 vcl7 vclA vclB
***  v1   CLI RX  200
**** v1   CLI RX|VCL vcl3 discarded
**** v1   CLI RX|VCL vclB discarded
**** v1   CLI RX|VCL vcl2 discarded
**** v1   CLI RX|VCL vcl4 discarded
**** v1   CLI RX|VCL vcl5 discarded
**** v1   CLI RX|VCL vcl6 discarded
**** v1   CLI RX|VCL vcl7 discarded
**** v1   CLI RX|VCL vclA discarded
**** v1   CLI RX|VCL vcl1 discarded
**   v1   CLI 200 <vcl.discard vcl1 vcl2 vcl3 vcl4 vcl5 vcl6 vcl7 vclA vclB>
```

Reply:
ah ok. So I agree that the vcls used for `vclA` returns could be named. But It would appear helpful in `vcl.list`

Reply:
But you can hardly do that in the tabular output, it would only work for the json output. That's why I kept `vcl.deps` in my patch series.

Reply:
ok.
but vcl.list should still show the fact, e.g. `<- (x return(NAME)'s)`